.github/workflows/terraform-compatibility-matrix.yml

```diff
@@ -2,8 +2,10 @@ name: "HashiCorp Terraform Compatibility Matrix"
 run-name: 'HashiCorp Terraform Compatibility Matrix ${{ inputs.atlas_cloud_env }}'
 
 on:
+  pull_request:
+    types: [synchronize]
   schedule:
-    - cron: "0 0 1-7 * 6" # runs first Saturday of the month at midnight UTC
```

Comment:
[POSIX cron syntax](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07): _"if either the month or day of month is specified as an element or list, and the day of week is also specified as an element or list, then any day matching either the month and day of month, or the day of week, shall be matched."_
This causes for the cron to be triggered each Saturday and all days between 1 and 7.